doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+==============================================
+Storage devices and OSD’s management workflows
+==============================================
+The cluster storage devices are the physical storage devices installed in each of the cluster’s hosts. We need to execute different operations over them and also to retrieve information about physical features and working behaviour.
+The basic use cases we have in this area are:
+
+- `1. Retrieve device information. Inventory`_
+- `2. Add OSDs`_
+- `3. Remove OSDs`_
+
+1. Retrieve device information. Inventory
+=========================================
+We must be able to review what is the current state and condition of the cluster storage devices. We need the identification and features detail(including ident/fault led on/off capable) and if the device is used or not as an OSD/DB/WAL device.
+
+The information required for each device will be at least:
+::
+
+    Hostname   Path      Type  Serial    Size   Health  Ident  Fault  Available
+
+.. Note: A more optional extended view with other information fields could be also useful.
+
+In order to know what is the current condition of the device, we need to know what is the amount of storage used, the percentage of free space, the average number of iops and  the fault led state. Part of the operational information for each device has its source in the data gathered by Prometheus. Given that the monitoring stack has reached a very important role in the Ceph Dashboard, it seems sensible to start to think in the monitoring stack as a fundamental piece of the UI installed by default.  Once we have Prometheus as an “always present” source of information, it seems sensible to start to use part of the information about performance and operational state in all the places where it can be useful to the final user.
+
+Another important question around retrieving device information is “efficiency”. The information about devices can be critical in components like the orchestrator or the dashboard, because this information usually is used to take decisions.
+When we talk about efficiency we need to be sure that all the points are covered:
+
+#. Get the complete information for each device in the most fast way.
+#. All the information about all the devices in one host is accessible always immediately.
+#. The information is constantly updated in each host. A device failure or the addition of a new device must be reflected in less than 1 second.
```

Comment:
Just trying to express that we need a very quick update of the inventory ... 1 second could be the upper limit ... but not more.

Reply:
Note that this may not always be achievable, as the operation to gather this information may be expensive / long running. If that's the case, I think we should allow for cached operations to be a suitable alternative, provided that the user is aware of the fact that they're seeing data that might be outdated (e.g. by indicating when it was last updated).

Reply:
Ok, so there is no scientific measure behind this. If speed is of such a matter, we should think about a different model than a _cached pull_. The most reactive way would probably be a event based pub-sub model, however I think that's overkill for that purpose.

Nit:

I'd avoid using arbitrary numbers. This will set false expectations for outsiders. What about rephrasing it to:

```
A device failure or the addition of a new device must be detected in the smallest possible timeframe
```

Reply:
Removed explicit reference of 1 second.

What is clear is that we agree about the importance of efficiency and scalability.